pkg/cmd/server/admin/create_signercert.go

```diff
@@ -99,9 +96,9 @@ func (o CreateSignerCertOptions) CreateSignerCert() (*crypto.CA, error) {
 		ca, written, err = crypto.EnsureCA(o.CertFile, o.KeyFile, o.SerialFile, o.Name)
 	}
 	if written {
-		fmt.Fprintf(o.Output.Get(), "Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
+		glog.V(3).Infof("Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
```

Comment:
If your code needs to not have this output, pass in a writer that discards the output.

You basically have two options. Pass down something so that the local routine dispatches the output correctly. Or have the local routine pass up the information to something that has the context for whether to output it. It's kinda ugly either way, but I think the existing way is cleanest. You have to pass things down regardless. Aggregating return values up the stack is yecch.